src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
```

Comment:
PullProofAI

It's recommended to use strict comparison (`===`) instead of loose comparison (`==`) for comparing `requestSecret` and `secret` to avoid unexpected type coercion issues.
```diff
-     if (requestSecret && requestSecret == secret) {
+     if (requestSecret && requestSecret === secret) {
```


<!-- This is an auto-generated comment by PullProofAI -->